### PR TITLE
Change search

### PR DESCRIPTION
**Pull request by matiasbenary**
That is done:
The search works for docs and for catalog.
When you click on docs it takes you inside the page.
Within apps when you click on the title it takes you where it belongs, you can also click on the tag. 

Missing:
Missing components, I don't quite understand the filters I was using.
Separate in component and improve the styles.
Missing the full view of search when you search in mobile.
When you click to close
Paginated and see all
Order the code

### CODE REVIEW

**Comment by gagdiez** [comment]
@shelegdmitriy we might need some help with css here, is that something you could help with? otherwise I'll see to dust off my css foo

cc @charleslavon

**Comment by shelegdmitriy** [comment]
> @shelegdmitriy we might need some help with css here, is that something you could help with? otherwise I'll see to dust off my css foo
> 
> cc @charleslavon

Sure I can help you here! Let me check the PR and I'll respond in a while

**Comment by shelegdmitriy** [comment]
@matiasbenary may I ask you to fix CI/Lint please?

**Comment by matiasbenary** [comment]
> @matiasbenary may I ask you to fix CI/Lint please?

I fixed it

File: src/utils/angoliaSearchApi.ts
Comment (denbite) [inline src/utils/angoliaSearchApi.ts:1]: typo -> `algoliaSearchApi.ts`

**Review by shelegdmitriy — CHANGES_REQUESTED** [review]
Знімок екрана 2024-07-22 о 21 33 04
Знімок екрана 2024-07-22 о 21 44 07

Clicking on -> arrow on the "Components" tab leads to error

cc @matiasbenary @gagdiez

**Review by shelegdmitriy — CHANGES_REQUESTED** [review]
#### Issue 2
Знімок екрана 2024-07-22 о 21 37 01

Search input is broken if sidebar is collapsed. On dev.near.org it works fine:

Знімок екрана 2024-07-22 о 21 50 15

**Comment by shelegdmitriy** [comment]
Знімок екрана 2024-07-22 о 21 36 47

I thought we decided to show this search input with dropdown for desktop devices only? I'd propose to stick to the current implementation: mobile only have a search icon which leads to search page. If you wish to improve this later - create a task.

cc @gagdiez

**Comment by shelegdmitriy** [comment]
Hey @matiasbenary and @gagdiez. I see you did a lot of progress here. There is a room for improvements and I would like to help you but may I ask you to fix this [issue 1](https://github.com/near/near-discovery/pull/1249#pullrequestreview-2192424151) please?

**Comment by shelegdmitriy** [comment]
Also, @matiasbenary ping me please after you fix issue 1 please so I could pull your branch to make some additional changes, ok?

**Comment by gagdiez** [comment]
I think the issue 1 is because we are soft-loading the page through a router, and not hard-refreshing the page. This keeps the state, and confuses the VM sometimes @matiasbenary

**Review by gagdiez — APPROVED** [review]

**Comment by gagdiez** [comment]
@shelegdmitriy while I was not able to reproduce the error that you and @matiasbenary was able to reproduce it in the current version of dev.near.org

I think it is related to another issue, which is how we handle components and the VM in general: https://github.com/near/near-discovery/issues/1264

I would suggest to leave this for another time, as it is an already existing bug

**Comment by shelegdmitriy** [comment]
Thank you @gagdiez. Let me add one more commit this this PR please and we'll good to go

**Comment by gagdiez** [comment]
before mergin, @charleslavon could you check if you get the same error as @shelegdmitriy ?, see comment above: https://github.com/near/near-discovery/pull/1249#pullrequestreview-2192424151

**Comment by shelegdmitriy** [comment]
Hey @gagdiez @matiasbenary @charleslavon. I believe I added everything I'd like to this PR so feel free to review this and leave your questions. One thing I want to highlight is new `<Input />` component. Actually, it was implemented by @calebjacob [here](https://github.com/near/near-discovery-components/blob/develop/src/DIG/Input.jsx) and polished in our ticketing repo so I copied that because it already has a lot of we need (and maybe even more) and it's fully aligned with current color scheme. This means, I would recommend you to use it in your future refactoring rounds. 

FYI: a lot of components are already implemented [here](https://github.com/near/near-discovery-components/tree/develop/src/DIG) and there was a preview page for all of them but I can't find it.

File: src/hooks/useGatewayEvents.ts
Comment (shelegdmitriy) [inline src/hooks/useGatewayEvents.ts:36]: @gagdiez why do you remove gleap from here?

**Review by shelegdmitriy — APPROVED** [review]
LGTM!

**Comment by gagdiez** [comment]
@race-of-sloths I summon you!

**Comment by race-of-sloths** [comment]
### 🙁 This repository is not a part of the [Race of Sloths](https://race-of-sloths.com)
Please ask the maintainer to [contact us](https://github.com/NEAR-DevHub/race-of-sloths/issues/new/choose) to join the race!

<details><summary><strong>What is the Race of Sloths</strong></summary>
<p>

Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag us inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/matiasbenary)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!

</p>
</details>

**Pull request merged**
